ledger/ledger.go

```diff
@@ -200,19 +207,24 @@ func FormatLedger(currency string, locale string, entries []Entry) (string, erro
 				al++
 			}
 			co <- struct {
+				i int
 				s string
 				e error
-			}{s: d + strings.Repeat(" ", 10-len(d)) + " | " + de + " | " +
+			}{i: i, s: d + strings.Repeat(" ", 10-len(d)) + " | " + de + " | " +
 				strings.Repeat(" ", 13-al) + a + "\n"}
-		}(et)
+		}(i, et)
 	}
+	ss := make([]string, len(entriesCopy))
 	for _ = range entriesCopy {
 		v := <-co
 		if v.e != nil {
 			return "", v.e
 		} else {
-			s += v.s
+			ss[v.i] = v.s
 		}
 	}
+	for i := 0; i < len(entriesCopy); i++ {
```

Comment:
the initial code is supposed to be badly written so I intentionally am not using `for _, s := range ss` here. Be assured that I would not normally write code like this...